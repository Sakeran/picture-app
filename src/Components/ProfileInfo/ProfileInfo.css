.ProfileInfo{
  background-color: white;
  margin-bottom: 1em;
  text-align: center;
}

.ProfileInfo-name {
  margin-top: 1em;
}

.ProfileInfo-section {
  margin: 1em 0;
}
