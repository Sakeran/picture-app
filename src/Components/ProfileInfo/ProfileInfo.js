import React from 'react';
import PropTypes from 'prop-types';

import './ProfileInfo.css';

const ProfileInfo = ({ name, location, bio}) => (
  <div className="ProfileInfo section-border">
    <h2 className="ProfileInfo-name">{name || "No Name Given"}</h2>
    <div className="ProfileInfo-section">
      <strong>Location:</strong> {location}
    </div>
    <div className="ProfileInfo-section">
      <strong>About:</strong> {bio}
    </div>
  </div>
);

ProfileInfo.propTypes = {
  name: PropTypes.string,
  location: PropTypes.string,
  bio: PropTypes.string,
};

export default ProfileInfo;
