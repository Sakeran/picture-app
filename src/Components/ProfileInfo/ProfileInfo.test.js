import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ProfileInfo from './ProfileInfo';

test('ProfileInfo renders correctly', () => {
  const props = {
    name: 'Test User',
    location: 'Somewhere',
    bio: 'A test user.'
  };
  const component = shallow(<ProfileInfo {...props} />);
  expect(component.text()).toContain('Test User');
  expect(component.text()).toContain('Somewhere');
  expect(component.text()).toContain('A test user');
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
