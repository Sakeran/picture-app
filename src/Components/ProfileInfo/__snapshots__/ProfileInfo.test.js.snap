// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileInfo renders correctly 1`] = `
<div
  className="ProfileInfo section-border"
>
  <h2
    className="ProfileInfo-name"
  >
    Test User
  </h2>
  <div
    className="ProfileInfo-row"
  >
    <h3>
      Location:
    </h3>
    <span
      className="ProfileInfo-span"
    >
      Somewhere
    </span>
  </div>
  <div
    className="ProfileInfo-row"
  >
    <h3>
      About Me:
    </h3>
    <span
      className="ProfileInfo-span"
    >
      A test user.
    </span>
  </div>
</div>
`;
